### DISCUSSION TITLE
Generate one page with programmatic API

### DISCUSSION THREAD

**Discussion by mark-reason** [11ty/eleventy#2674] [Q&A]
Is there a way to use the Eleventy programmatic API to generate just one page and also feed in the data it uses at run time?

**Comment by pdehaan** [comment]
This worked for me locally:
```js
// build.js
const Eleventy = require("@11ty/eleventy");

(async function() {
  let elev = new Eleventy("src/pages/two.liquid");
  await elev.write();
})();
```

Ref: https://www.11ty.dev/docs/programmatic/#changing-the-input-and-output-directories

Where my fake site looks like this:
```sh
tree -aI node_modules
.
├── .eleventy.js
├── build.js
├── package-lock.json
├── package.json
├── src/
│   ├── index.liquid
│   └── pages/
│       ├── one.liquid
│       ├── three.liquid
│       └── two.liquid
└── www/
    └── two/
        └── index.html

4 directories, 9 files
```

And the output looks like:
```sh
npm run build

> 11ty-2674@1.0.0 build
> node build

building...
[11ty] Writing www/two/index.html from ./src/pages/two.liquid
```

And if I do a full build (using the CLI) it looks like this:
```sh
npx @11ty/eleventy
building...
[11ty] Writing www/index.html from ./src/index.liquid
[11ty] Writing www/pages/one/index.html from ./src/pages/one.liquid
[11ty] Writing www/pages/three/index.html from ./src/pages/three.liquid
[11ty] Writing www/pages/two/index.html from ./src/pages/two.liquid
[11ty] Wrote 4 files in 0.02 seconds (v1.0.2)
```

&hellip; or, you can use the CLI to build a single page using the `--input` flag:
```sh
npx @11ty/eleventy --input src/pages/three.liquid
building...
[11ty] Writing www/three/index.html from ./src/pages/three.liquid
[11ty] Wrote 1 file in 0.02 seconds (v1.0.2)
```

**Reply by mark-reason** [reply]
Sorry I should have been more specific @pdehaan.

So I've got a paginated template that runs through events data to produce event pages. However I will also have user submitted events which will be added to the event data and I want to be able to view the new user submitted event page without triggering a rebuild, would something like that be possible?

**Reply by pdehaan** [reply]
Not that I can immediately think of, unless you can get clever with the [`before` callback](https://www.11ty.dev/docs/pagination/#the-before-callback).

**Reply by mark-reason** [reply]
I did attempt a solution using the serverless plugin as it mentions it can be used for dynamic content like page previews but I ran into the following issue: https://github.com/11ty/eleventy/issues/2667
